dotnet/src/Functions/Functions.OpenAPI/OpenAI/KernelOpenAIPluginExtensions.cs

```diff
@@ -195,7 +202,7 @@ public static class KernelOpenAIPluginExtensions
         try
         {
             pluginJson = JsonNode.Parse(openAIManifest)!;
-            openAIAuthConfig = pluginJson["auth"].Deserialize<OpenAIAuthenticationConfig>()!;
+            openAIAuthConfig = pluginJson["auth"].Deserialize<OpenAIAuthenticationConfig>(s_jsonOptionsOpenAIManifest)!;
```

Comment:
I'm a little confused... which members is this snake casing meant to influence?  Most of the members I see have an explicit JsonPropertyName, and this change is also removing an enum and replacing it with a string.

Reply:
Thanks for this question Stephen.  The high-order bit here is that our OpenAI function calling story is functionally broken.  This is a priority fix.

This issue at play has to do with the enums: `OpenAIAuthenticationType` and `OpenAIAuthorizationType`.  This is where the snake casing on deserialization was causing issue (re: stack trace).

The reason I removed the third enum is that is just seems silly to define mime/type as enum and then attempt to marshall value conversion...when mime-type is normally adequately handled as string.  Also, if a new mime-type is added...it won't break enum parsing.  (Apparently EnumMemberAttribute doesn't impact json deserialization and this code path was never tested.)

Reply:
> Apparently EnumMemberAttribute doesn't impact json deserialization

It doesn't impact JsonSerializer, period :) It's part of an entirely different serializer.  All usage here should be removed.